### PR TITLE
feat(color): add `FromStr` implementation for `Color`

### PR DESCRIPTION
**Pull request by Mehrbod2002**
fixes: #143

### CODE REVIEW

**Review by joshka — CHANGES_REQUESTED** [review]
There's a note in the issue about whether we should / shouldn't bring in a new dependency for this (it seems like a fairly small amount of code to avoid doing so). I could go either way, but lean towards not bringing it in. Do we gain anything by putting the implementation off onto colorsys?

File: src/style.rs [hunk 1]
```diff
@@ -312,3 +314,49 @@ mod tests {
...
+            "lightred" | "light red" => Self::LightRed,
+            "lightgreen" | "light green" => Self::LightGreen,
+            "lightyellow" | "light yellow" => Self::LightYellow,
+            "lightblue" | "light blue" => Self::LightBlue,
+            "lightmagenta" | "light magenta" => Self::LightMagenta,
+            "lightcyan" | "light cyan" => Self::LightCyan,
+            "white" => Self::White,
+            _ => {
+                if let Ok(rgb) = Rgb::from_hex_str(&format!("#{s}")) {
+                    Self::Rgb(rgb.red() as u8, rgb.green() as u8, rgb.blue() as u8)
```
[the remark this answers is not in the document]
Comment (Mehrbod2002) [reply src/style.rs]: but I should think without #

**Review by sophacles — CHANGES_REQUESTED** [review]
Code looks good to me.

Would like to see documentation - its non-obvious how to make Indexed and Rgb from a string without reading the code.

**Comment by orhun** [comment]
I think we should also add unit tests for this.

**Comment by Mehrbod2002** [comment]
> I think we should also add unit tests for this.

The tests added

**Review by joshka — APPROVED** [review]
Nice work. This looks good to me.

**Review by sophacles — DISMISSED** [review]

File: src/style.rs [hunk 2]
```diff
@@ -311,4 +386,28 @@ mod tests {
...
+    fn test_custom_color() {
+        let color: Color = Color::from_str("lightblue").unwrap();
+        assert_eq!(color, Color::LightBlue);
+    }
+
+    #[test]
+    fn test_invalid_color() {
+        let color: Result<Color, _> = Color::from_str("invalid_color");
+        assert!(color.is_err());
+    }
```
Comment (sophacles) [inline src/style.rs:412]: Modifying this test to: 
```rust
    #[test]
    fn test_invalid_colors() {
        let bad_colors = [
            "invalid_color", // not a color string
            "abcdef0",       // 7 chars is not a color
            " bcdefa",       // doesn't start with a '#'
            "blue ",         // has space at end
            " blue",         // has space at start
            "#abcdef00",     // too many chars
        ];

        for bad_color in bad_colors {
            assert!(
                Color::from_str(bad_color).is_err(),
                "bad color: '{}'",
                bad_color
            );
        }
    }
```
Fails

File: src/style.rs [hunk 3]
```diff
@@ -246,6 +247,80 @@ impl Style {
...
+            "lightgreen" | "light green" => Self::LightGreen,
+            "lightyellow" | "light yellow" => Self::LightYellow,
+            "lightblue" | "light blue" => Self::LightBlue,
+            "lightmagenta" | "light magenta" => Self::LightMagenta,
+            "lightcyan" | "light cyan" => Self::LightCyan,
+            "white" => Self::White,
+            _ => {
+                if let Ok(index) = s.parse::<u8>() {
+                    Self::Indexed(index)
+                } else if let (Ok(r), Ok(g), Ok(b)) = (
```
Comment (sophacles) [inline src/style.rs:310]: If we do something like this (my code here is ugly, but gets the job done):
```rust
                } else if let (Ok(r), Ok(g), Ok(b)) = {
                    if !s.starts_with('#') || s.len() != 7 {
                        return Err(ParseColorError);
                    }
                    (
                        u8::from_str_radix(&s[1..3], 16),
                        u8::from_str_radix(&s[3..5], 16),
                        u8::from_str_radix(&s[5..7], 16),
                    )
                } {
                    Self::Rgb(r, g, b)
```
The test cases below pass.

**Review by sophacles — CHANGES_REQUESTED** [review]
There's some edge cases that we probably want to deal with.

**Review by sophacles — APPROVED** [review]

**Review by orhun — APPROVED** [review]
Neat.
